client/lib/analytics/ad-tracking/woo/load-woo-gtm-container.ts

```diff
@@ -0,0 +1,46 @@
+import { loadScript } from '@automattic/load-script';
+import { mayWeTrackByTracker } from '../../tracker-buckets';
+import { GOOGLE_GTM_SCRIPT_URL, TRACKING_IDS } from '../constants';
+
+/**
+ * We expect GTM to need these globals to be declared.
+ */
+declare global {
+	interface Window {
+		dataLayer: {
+			push: ( data: Record< string, unknown > ) => void;
+		};
+		google_tag_manager: Record< string, unknown >;
+	}
+}
+
+/**
+ * Loads the GTM script with the Woo container ID, if the user has consented to tracking,
+ * and tracking is allowed by the current environment.
+ *
+ * @returns Promise<void>
+ */
+export const loadWooGTMContainer = async (): Promise< void > => {
+	// Is Google Tag Manager already initialized?
+	if ( window.dataLayer && window.google_tag_manager ) {
+		return;
+	}
+
+	// Are we allowed to track (user consent, e2e, etc.)?
+	if ( ! mayWeTrackByTracker( 'googleTagManager' ) ) {
+		return;
+	}
+
+	// Load the Google Tag Manager script
+	await loadScript( GOOGLE_GTM_SCRIPT_URL + TRACKING_IDS.wooGoogleTagManagerId );
+};
+
+/**
+ * Initializes the GTM container.
+ *
+ * @returns Promise<void>
+ */
+export const initWooGTMContainer = async (): Promise< void > => {
+	window.dataLayer = window.dataLayer || [];
```

Comment:
`dataLayer` sounds like an odd name for this, and this could be easily confused with the existing Calypso data layer:

https://github.com/Automattic/wp-calypso/blob/5f4eb761353006b94555609175101ac4f95d7ac1/client/state/data-layer/README.md#L1

Reply:
That's actually what Google bases their implementation on: https://developers.google.com/tag-platform/tag-manager/web/datalayer#installation

They require initialization of `dataLayer` property in `window` property as they access it from their scripts.

Reply:
Makes sense. We could add some comment, indicating something like "not to be confused with the Calypso Data Layer".

Reply:
Right, that would help, indeed!